lib/twilio/device.ts

```diff
@@ -1014,162 +1027,168 @@ class Device extends EventEmitter {
    * @param twimlParams - A flat object containing key:value pairs to be sent to the TwiML app.
    * @param options - Options to be used to instantiate the {@link Call}.
    */
-  private async _makeCall(twimlParams: Record<string, string>, options?: Call.Options, isReconnect: boolean = false): Promise<Call> {
-    if (typeof Device._isUnifiedPlanDefault === 'undefined') {
-      throw new InvalidStateError('Device has not been initialized.');
-    }
+  private _makeCall(twimlParams: Record<string, string>, options?: Call.Options, isReconnect: boolean = false): Promise<Call> {
```

Comment:
Instead of wrapping the whole _makeCall method, I was thinking on line 609, you can do this
```ts
this._makeCallPromise = this._makeCall(twimlParams, callOptions, isReconnect);
const activeCall = this._activeCall = await this._makeCallPromise;
```
Also, can you check if we also need this inside onSignalingInvite?

Reply:
Then wherever we're setting `this._activeCall` to null, we should also set `this._makeCallPromise` to null.

Reply:
@kpchoy I think you missed the last comment above ^

Reply:
Oops sorry @charliesantos , added that in. Please take another look.